spec/git-repository-async-spec.js

```diff
@@ -0,0 +1,598 @@
+    })
+
+    it('returns the diff stat', async () => {
+      const filePath = path.join(workingDirectory, 'a.txt')
+      fs.writeFileSync(filePath, 'change')
+
+      const {added, deleted} = await repo.getDiffStats('a.txt')
+      expect(added).toBe(1)
+      expect(deleted).toBe(0)
+    })
+  })
+
+  describe('.hasBranch(branch)', () => {
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+    })
+
+    it('resolves true when the branch exists', async () => {
+      const hasBranch = await repo.hasBranch('master')
+      expect(hasBranch).toBeTruthy()
+    })
+
+    it("resolves false when the branch doesn't exist", async () => {
+      const hasBranch = await repo.hasBranch('trolleybus')
+      expect(hasBranch).toBeFalsy()
+    })
+  })
+
+  describe('.getReferences(path)', () => {
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+    })
+
+    it('returns the heads, remotes, and tags', async () => {
+      const {heads, remotes, tags} = await repo.getReferences()
+      expect(heads.length).toBe(1)
+      expect(remotes.length).toBe(0)
+      expect(tags.length).toBe(0)
+    })
+  })
+
+  describe('.getReferenceTarget(reference, path)', () => {
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+    })
+
+    it('returns the SHA target', async () => {
+      const SHA = await repo.getReferenceTarget('refs/heads/master')
+      expect(SHA).toBe('8a9c86f1cb1f14b8f436eb91f4b052c8802ca99e')
+    })
+  })
+
+  describe('.getConfigValue(key, path)', () => {
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+    })
+
+    it('looks up the value for the key', async () => {
+      const bare = await repo.getConfigValue('core.bare')
+      expect(bare).toBe('false')
+    })
+
+    it("resolves to null if there's no value", async () => {
+      const value = await repo.getConfigValue('my.special.key')
+      expect(value).toBeNull()
+    })
+  })
+
+  describe('.checkoutReference(reference, create)', () => {
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+    })
+
+    it('can create new branches', () => {
+      let success = false
+      let threw = false
+      waitsForPromise(() => repo.checkoutReference('my-b', true)
+        .then(_ => success = true)
+        .catch(_ => threw = true))
+      runs(() => {
+        expect(success).toBeTruthy()
+        expect(threw).toBeFalsy()
+      })
+    })
+  })
+
+  describe('.getLineDiffs(path, text)', () => {
+    beforeEach(() => {
+      const workingDirectory = copyRepository()
+      repo = GitRepositoryAsync.open(workingDirectory)
+    })
+
+    it('can get the line diff', async () => {
```

Comment:
Maybe something like "returns the old and new lines of the diff" would be a clearer name for this spec.